libass/ass_utils.h

```diff
@@ -55,6 +55,18 @@
 
 #define FEATURE_MASK(feat) (((uint32_t) 1) << (feat))
 
+#ifndef __has_builtin
+    #define __has_builtin(x) 0
+#endif
+
+#if __has_builtin(__builtin_unreachable)
```

Comment:
```suggestion
#if __has_builtin(__builtin_assume)
    #define ASSUME(x) __builtin_assume(x)
#elif __has_builtin(__builtin_unreachable)
```

Reply:
Applied in modified form (preferring MSVC's `__assume` over the `__builtin_unreachable` version). Note the distinction: `__builtin_assume` and `__assume` are guaranteed never to emit code for the conditional, while `if (x) __builtin_reachable()` may if the optimizer's feeling dumb (maybe on `-O0`).